tools/go-agent/tools/types.go

```diff
@@ -17,10 +17,19 @@
 
 package tools
 
+var basicDataTypes = make(map[string]bool)
```

Comment:
I suggest using `map[string]struct{}` as it is less space but also to declare the map in the var instead of initializing on init.